demisto_sdk/commands/validate/validators/BC_validators/BC107_is_valid_toversion_on_modified.py

```diff
@@ -14,30 +16,111 @@
+        GitStatuses.MODIFIED,
+        GitStatuses.RENAMED,
+        GitStatuses.ADDED,
+    ]
+    valid_cases = (
+        "The old item `fromversion` field should be less than the new item `fromversion` field\n"
+        "The old item `toversion` field should be less than the new item `fromversion` field\n"
+        "The old and the new item should be continuous, aka the old one `toversion` is one version"
+        " less than the new one `fromversion`"
+    )
 
     def obtain_invalid_content_items(
         self, content_items: Iterable[ContentTypes]
     ) -> List[ValidationResult]:
+        modified_items, new_items = sort_content_items(content_items)
         return [
             ValidationResult(
                 validator=self,
-                message=self.error_message,
+                message=self.error_message.format(
+                    content_item.object_id, self.valid_cases
+                ),
                 content_object=content_item,
             )
-            for content_item in content_items
-            if toversion_modified(content_item)
+            for content_item in modified_items
+            if self.invalid_toversion(content_item, new_items)
         ]
 
+    def invalid_toversion(self, modified_item: ContentTypes, new_items: dict):
+        old_file = cast(ContentTypes, modified_item.old_base_content_object)
+        if modified_item.toversion != old_file.toversion:  # file toversion was updated
+            if modified_item.object_id in new_items:
+                new_item = new_items[modified_item.object_id]
+                if not is_valid_versions(
+                    Version(modified_item.fromversion),
+                    Version(modified_item.toversion),
+                    Version(new_item.get("from")),
+                ):
+                    return True
+            else:
+                return True
 
-def toversion_modified(content_item: ContentTypes) -> bool:
-    if not content_item.old_base_content_object:
         return False
-    old_file = cast(ContentTypes, content_item.old_base_content_object)
-    return content_item.toversion != old_file.toversion
+
+
+def sort_content_items(content_items: Iterable[ContentTypes]):
+    """
+    Sort Content Items into two lists of Modified content items and newly added content items.
+
+    Args:
+        content_items (Iterable[ContentTypes]): The content items list
+
+    Returns:
+        modified_items: modified items.
+        new_items: newly added items.
+    """
+    modified_items: list = []
+    new_items: dict = {}
+    for item in content_items:
+        if item.git_status in [GitStatuses.MODIFIED, GitStatuses.RENAMED]:
+            modified_items.append(item)
+        elif item.git_status == GitStatuses.ADDED:
+            new_items[item.object_id] = {"from": item.fromversion, "to": item.toversion}
+
+    return modified_items, new_items
+
+
+def is_valid_versions(old_from: Version, old_to: Version, new_from: Version) -> bool:
```

Comment:
please try to have more meaningful/accurate names in your methods - is_valid_versions is very vague, you have to read the code to understand your definition of what is valid or not.
Specifically here, we defined a rule to describe "continuousness of versions" which I'm not sure it's always accurate - for example, what about the jump between XSOAR 6 and 8?